packages/lodestar/src/util/httpClient.ts

```diff
@@ -0,0 +1,49 @@
+import Axios, {AxiosResponse, AxiosInstance} from "axios";
+
+export class HttpClient {
+  private _client: AxiosInstance;
+
+  public constructor(urlPrefix?: string) {
+    const _urlPrefix = urlPrefix? urlPrefix : "";
+    this._client = Axios.create({
+      baseURL: _urlPrefix
+    });
+  }
+
+  public get<T>(url: string): Promise<T> {
+    return new Promise((resolve, reject) => {
+      this._client.get<T>(url).then((result: AxiosResponse<T>) => {
+        resolve(result.data);
+      }).catch((reason: any) => {
+        handleError(reason, reject);
+      });
+    });
+  }
+
+  public post<T>(url: string, data: T): Promise<any> {
+    return new Promise((resolve, reject) => {
+      this._client.post(url, data).then((result: AxiosResponse<any>) => {
+        resolve(result.data);
+      }).catch((reason: any) => {
+        handleError(reason, reject);
+      });
+    });
+  }
+}
+
+const handleError = (reason, reject) => {
```

Comment:
add types

Reply:
`reason` is from `try/catch`, `reject` is from Promise so I find no type for them.

Reply:
reason is probably string, rject is Function as it accepts any param

Reply:
- reject is Function - but you don't really need it
- reason is actually error object with type [AxiosError](https://github.com/axios/axios/blob/master/index.d.ts#L79)

You should implement error checking like this: https://github.com/axios/axios#handling-errors so logs can tell us it was client error, server or transport error